vllm/config.py

```diff
@@ -191,6 +192,18 @@ def __init__(
                    f"'Please instead use `--hf-overrides '{hf_override!r}'`")
             warnings.warn(DeprecationWarning(msg), stacklevel=2)
 
+        if is_s3(model):
```

Comment:
In addition to weights files, to load model we need `config.json` and tokeinzer files.
If they are stored in s3 as well, the program needs to read it from there. Which before this change, is not possible.

There are 3 options for implementation for this:

![image](https://github.com/user-attachments/assets/7ffb96a4-4043-456f-aeac-76ddef1b63d6)
(This image shows in high level the relevant chain of calls in the code)

Implement it like option 1, in the `vllm/config.json` means a single place of change, small as possible.
Option 2 means implement it in 2 files, seperately, in the tokeinzer and the config file. One may argue its prefered way, as model scope hub integrration is in this layer.
Option 3 means, lets implement it in HuggingFace library, lets make it able to get path from s3 bucket, and read the content from it.

In my opinion option 3 is the most transparent one and the most right one, however they are not interested in expanding their support (https://github.com/huggingface/transformers/issues/19834#issuecomment-1289079562)